kanidm_unix_int/src/daemon.rs

```diff
@@ -380,302 +379,313 @@ async fn main() {
-                cfg_path_str
-            );
-        }
-    }
-
-    let unixd_path = Path::new(DEFAULT_CONFIG_PATH);
-    let unixd_path_str = match unixd_path.to_str() {
-        Some(cps) => cps,
-        None => {
-            error!("Unable to turn unixd_path to str");
-            std::process::exit(1);
-        }
-    };
-    if !unixd_path.exists() {
-        // there's no point trying to start up if we can't read a usable config!
-        error!(
-            "unixd config missing from {} - cannot start up. Quitting.",
-            unixd_path_str
-        );
-        std::process::exit(1);
-    } else {
-        let unixd_meta = match metadata(&unixd_path) {
-            Ok(v) => v,
-            Err(e) => {
-                error!("Unable to read metadata for {} - {:?}", unixd_path_str, e);
-                std::process::exit(1);
-            }
-        };
-        if !file_permissions_readonly(&unixd_meta) {
-            warn!("permissions on {} may not be secure. Should be readonly to running uid. This could be a security risk ...",
-                unixd_path_str);
-        }
-
-        if unixd_meta.uid() == cuid || unixd_meta.uid() == ceuid {
-            warn!("WARNING: {} owned by the current uid, which may allow file permission changes. This could be a security risk ...",
-                unixd_path_str
-            );
-        }
-    }
-
-    // setup
-    let cb = match KanidmClientBuilder::new().read_options_from_optional_config(cfg_path) {
-        Ok(v) => v,
-        Err(_) => {
-            error!("Failed to parse {}", cfg_path_str);
-            std::process::exit(1);
-        }
-    };
-
-    let cfg = match KanidmUnixdConfig::new().read_options_from_optional_config(unixd_path) {
-        Ok(v) => v,
-        Err(_) => {
-            error!("Failed to parse {}", unixd_path_str);
-            std::process::exit(1);
-        }
-    };
-
-    debug!("🧹 Cleaning up sockets from previous invocations");
-    rm_if_exist(cfg.sock_path.as_str());
-    rm_if_exist(cfg.task_sock_path.as_str());
-
-    let cb = cb.connect_timeout(cfg.conn_timeout);
-
-    let rsclient = match cb.build() {
-        Ok(rsc) => rsc,
-        Err(_e) => {
-            error!("Failed to build async client");
-            std::process::exit(1);
-        }
-    };
-
-    // Check the pb path will be okay.
-    if cfg.db_path != "" {
-        let db_path = PathBuf::from(cfg.db_path.as_str());
-        // We only need to check the parent folder path permissions as the db itself may not exist yet.
-        if let Some(db_parent_path) = db_path.parent() {
-            if !db_parent_path.exists() {
+    tracing_forest::worker_task()
+        .set_global(true)
+        // Fall back to stderr
+        .map_sender(|sender| sender.or_stderr())
+        .build_on(|subscriber| subscriber
+            .with(EnvFilter::try_from_default_env()
```

Comment:
does the logging config in `server.toml` do anything? it seems to have been been broken for a while...

Reply:
Part of the goal with the tagging stuff is to add that functionality back.

Reply:
Sweeeeet, just confirming :D